src/openmc_heat_driver.cpp

```diff
@@ -100,36 +125,143 @@ void OpenmcHeatDriver::init_mappings()
     }
   }
 
+  // check that all fissionable cells have been mapped to the T/H model.
+  // Note that this does not check for distortions in the model, just that
+  // the surrogate T/H model is sufficiently fine in the pellet region to
+  // capture all fissionable cells in OpenMC.
+  Ensures(openmc_driver_->n_fissionable_cells_ == n_fissionable_mapped);
+
+  // set the number of solid cells before defining mappings for fluid cells
   if (openmc_driver_->active()) {
-    n_materials_ = openmc_driver_->cells_.size();
+    n_solid_cells_ = openmc_driver_->cells_.size();
+  }
+
+  // Establish mappings between fluid regions and OpenMC cells; it is assumed that there
+  // are no azimuthal divisions in the fluid phase in the OpenMC model so that we
+  // can take a point on a 45 degree ray from the pin center. TODO: add a check to make
+  // sure that the T/H model is finer than the OpenMC model.
+  for (int i = 0; i < heat_driver_->n_pins_; ++i) {
+    double x_center = heat_driver_->pin_centers_(i, 0);
+    double y_center = heat_driver_->pin_centers_(i, 1);
+
+    for (int j = 0; j < heat_driver_->n_axial_; ++j) {
+      double zavg = 0.5 * (z(j) + z(j + 1));
+      double l = heat_driver_->pin_pitch() / std::sqrt(2.0);
+      double d = (l - heat_driver_->clad_outer_radius_) / 2.0;
+      double x = x_center + (heat_driver_->clad_outer_radius_ + d) * std::sqrt(2.0) / 2.0;
+      double y = y_center + (heat_driver_->clad_outer_radius_ + d) * std::sqrt(2.0) / 2.0;
+
+      // Determine cell instance corresponding to given fluid location
+      Position r{x, y, zavg};
+      CellInstance c{r};
+      if (tracked.find(c) == tracked.end()) {
+        openmc_driver_->cells_.push_back(c);
+        tracked[c] = openmc_driver_->cells_.size() - 1;
```

Comment:
`size()` might be too large, since the values of `tracked` are ints.  Maybe use `gsl::narrow` or `gsl::narrow_cast`?